### PR TITLE
fix(constructors): handle goog.inherits and emit constructors

### LINKED ISSUE #22: Producing constructor with return type
this input
```
goog.provide('a');

/**
 * @constructor
 */
a.One = function() {};
a.One.prototype.foo = function(x) {};

/**
 * @constructor
 * @extends {a.One}
 */
a.Two = function() {};

goog.inherits(a.Two, a.One);
```

produces a class Two which has an explicit constructor. We have a return type on the result which is invalid syntax.

```
declare module 'goog:a' {
  export class One {
    foo (a : any ) : void ;
  }
  export function foo (a : any ) : void ;
  export class Two {
    constructor ( ...a : any [] ) : any ;
    static superClass_ : Object ;
  }
}
```

### PR DESCRIPTION
**Pull request by alexeagle**
Fixes #22

### CODE REVIEW

File: src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java [hunk 1]
```diff
@@ -641,10 +655,15 @@ private void visitProperties(ObjectType objType, boolean isStatic) {
...
+      visitFunctionType(ftype);
+      JSType type = ftype.getReturnType();
+      if (type != null) {
+        emit(":");
+        visitType(type);
+      }
     }
 
-    private void visitFunctionType(FunctionType ftype, String separator) {
+    private void visitFunctionType(FunctionType ftype) {
```
Comment (rkirov) [inline src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java]: I find it semi-suprising that visiting function type doesn't take care of constructors, add a comment to remind ourselves.

Comment (rkirov) [inline src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java]: or rename to visitFunctionArgs.

**Comment by rkirov** [comment]
LGTM, just one naming nit.

File: src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java (hunk 1 above)
Comment (alexeagle) [inline src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java]: renamed to visitFunctionSignature

**Comment by alexeagle** [comment]
merged in e0a9f88

**Pull request closed without merging**